### PR TITLE
Enable docker build with custom version of HDF5.

### PR DESCRIPTION
**Pull request by gonuke**
Also clean up/streamline some of the build process.

I'm not sure we need this for CI in the long term(??), but it will definitely help test #1220

### CODE REVIEW

File: docker/ubuntu_18.04-dev.dockerfile [hunk 1]
```diff
@@ -58,50 +58,67 @@ RUN if [ "${py_version%.?}" -eq 3 ] ; \
...
+        && cd .. \
+        && rm -rf hdf5; \
+    fi
+# put MOAB on the path
+ENV LD_LIBRARY_PATH $HDF5_INSTALL_PATH/lib:$LD_LIBRARY_PATH
+ENV LIBRARY_PATH $HDF5_INSTALL_PATH/lib:$LIBRARY_PATH
+
+ARG build_moab=NO
+ARG enable_pymoab=NO
+ENV INSTALL_PATH=$HOME/opt/moab
```
Comment (bam241) [inline docker/ubuntu_18.04-dev.dockerfile:88]: I believe than ARG command must be placed just before line they are used, 
I am unsure if having a ENV in-between could be a problem.

you might need to redeclare `build_hdf5` as an argument there along with `build_moab` and `enable_moab`

Comment (gonuke) [reply docker/ubuntu_18.04-dev.dockerfile:88]: This is not a problem - everything works as it is....

File: .circleci/config.yml [hunk 2]
```diff
@@ -323,6 +343,14 @@ workflows:
           requires:
             - py3_dagmc_pymoab_build
 
+      - py3_dagmc_pymoab_hdf5_12_build:
```
Comment (gonuke) [inline .circleci/config.yml:346]: @bam241 How do I make it so that this only produces a warning - and not a failure?

**Comment by gonuke** [comment]
This is supposed to fail on HDF5 v1.12,  working on amending it to produce a warning instead of a failure

File: .circleci/config.yml (hunk 2 above)
Comment (gonuke) [reply .circleci/config.yml:346]: Or is that a Travis thing?

Comment (bam241) [reply .circleci/config.yml:346]: I don't think this exists in CircleCI.

but we can decide to block or not merge based on the results of some tests.

File: docker/ubuntu_18.04-dev.dockerfile (hunk 1 above)
Comment (bam241) [reply docker/ubuntu_18.04-dev.dockerfile:88]: I had to make sure...

I was convinced that during one of my previous experience with docker `ARGS` I add to re-add the `ARG MY_VAR=XX` before each command that was using the variable....

I tested and apparently not...

**Comment by bam241** [comment]
SO apparently allow failure is not a feature in CircleCI, 
but we can do the opposite: block merge if one of the other tests is failing
in the GitHub repo: `settings/Branches/"Branch protection rules"`

**Comment by gonuke** [comment]
The current test failure is expected and will be resolved by #1220.  I think we should merge this and then see how #1220 performs in CI.

**Comment by bam241** [comment]
sounds about right! merging !

**Comment by bam241** [comment]
thx @gonuke

**Pull request merged**